corporate/tests/test_stripe.py

```diff
@@ -2839,6 +2839,130 @@ def test_downgrade_during_invoicing(self, *mocks: Mock) -> None:
         self.assertIsNone(plan.next_invoice_date)
         self.assertEqual(plan.status, CustomerPlan.ENDED)
 
+    @mock_stripe()
+    def test_switch_now_free_trial_from_monthly_to_annual(self, *mocks: Mock) -> None:
+        user = self.example_user("hamlet")
+        self.login_user(user)
+
+        free_trial_end_date = self.now + timedelta(days=60)
+        with self.settings(FREE_TRIAL_DAYS=60):
+            with patch("corporate.lib.stripe.timezone_now", return_value=self.now):
+                self.add_card_and_upgrade(user, schedule="monthly")
+                plan = CustomerPlan.objects.get()
+                self.assertEqual(plan.next_invoice_date, free_trial_end_date)
+                self.assertEqual(get_realm("zulip").plan_type, Realm.PLAN_TYPE_STANDARD)
+                self.assertEqual(plan.status, CustomerPlan.FREE_TRIAL)
+
+                customer = get_customer_by_realm(user.realm)
+                assert customer is not None
+                with self.assertLogs("corporate.stripe", "INFO") as m:
+                    result = self.client_patch(
+                        "/json/billing/plan",
+                        {"status": CustomerPlan.SWITCH_NOW_FREE_TRIAL_FROM_MONTHLY_TO_ANNUAL_CYCLE},
+                    )
+                    expected_log = f"INFO:corporate.stripe:Change plan status: Customer.id: {customer.id}, CustomerPlan.id: {plan.id}, status: {CustomerPlan.SWITCH_NOW_FREE_TRIAL_FROM_MONTHLY_TO_ANNUAL_CYCLE}"
+                    self.assertEqual(m.output[0], expected_log)
+                self.assert_json_success(result)
+
+                plan.refresh_from_db()
+                self.assertEqual(
+                    plan.status, CustomerPlan.SWITCH_NOW_FREE_TRIAL_FROM_MONTHLY_TO_ANNUAL_CYCLE
+                )
+
+                update_license_ledger_if_needed(user.realm, self.now)
+
+                plan.refresh_from_db()
+                self.assertEqual(plan.status, CustomerPlan.ENDED)
+
+                plan = CustomerPlan.objects.get(
+                    customer=customer,
+                    automanage_licenses=True,
+                    price_per_license=8000,
+                    fixed_price=None,
+                    discount=None,
+                    billing_cycle_anchor=self.now,
+                    billing_schedule=CustomerPlan.ANNUAL,
+                    invoiced_through=None,
+                    next_invoice_date=free_trial_end_date,
+                    tier=CustomerPlan.STANDARD,
+                    status=CustomerPlan.FREE_TRIAL,
+                    charge_automatically=True,
+                )
```

Comment:
Should this assert the plan exists or something? The test is kinda weird but it looks like the intent is to confirm that the  current plan is `ANNUAL` and has all the other expected properties?

Reply:
Yeah, we are checking if it has all the required properties. We get an error too if anything doesn't match.

> corporate.models.CustomerPlan.DoesNotExist: CustomerPlan matching query does not exist

Reply:
OK, that works.